plugins/hypervisors/vmware/src/com/cloud/hypervisor/vmware/resource/VmwareResource.java

```diff
@@ -4879,10 +4889,16 @@ private VirtualMachineGuestOsIdentifier translateGuestOsIdentifier(String cpuArc
                         } else if (objProp.getName().contains(instanceNameCustomField)) {
```

Comment:
Here it is the same as in CitrixResourceBase.java. I know you just followed the way the code was written here before, but this “if/else/if/else……” structure is too big; I believe to introduce some changes like this here; it would be much appreciated a refactoring. I do not mean a huge refactoring, only the extraction of this if/else structure to a method and then using some other clever way to do these checks and to retrieve the values.